### DISCUSSION TITLE
Is it possible to call other LLM endpoint like a self-hosted CloudGPT?

### DISCUSSION THREAD

**Discussion by daviddwlee84** [microsoft/promptflow#2340] [Q&A] [answered]
Seems Prompt Flow currently only supports OpenAI and Azure OpenAI natively.
Is it possible to work with other self-hosted LLM API endpoints such as CloudGPT?
https://github.com/noahweber1/cloudGPT

**Comment by brynn-code** [comment, marked as answer]
Hi, promptflow supports user call any endpoints in their own python script using script tool, and we have a 'CustomConnection' type for user to store their custom keys. Docs about [define a python script tool](https://microsoft.github.io/promptflow/how-to-guides/develop-a-flow/develop-standard-flow.html#define-python-node-interface).

Further on, we also supports user create there own customized llm tool for better usage and sharing, please reach the docs here for more details about [customizing an llm tool](https://microsoft.github.io/promptflow/how-to-guides/develop-a-tool/customize_an_llm_tool.html#customizing-an-llm-tool).
